proxy/fastcgi/fcgiclient.go

```diff
@@ -0,0 +1,581 @@
+// Copyright 2015 Light Code Labs, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Forked Jan. 2018 from https://github.com/mholt/caddy/blob/master/caddyhttp/fastcgi/fcgiclient.go
```

Comment:
I'd actually like to see this as a separate commit with the commit sha of the fork in the commit msg, e.g.

`Forking <sha> of https://github.com/mholt/caddy/...`

Pls update the `NOTICE` file as well.

I'm wondering whether we cannot just vendor this file in. Would require some tweaking but then it is clear where this is coming from.

Reply:
I'll break it down into separate commits when I'm done with changes.